### PR TITLE
Improvement of supplicant

### PR DESCRIPTION
**Pull request by unizippro**

### CODE REVIEW

File: ublox-short-range/src/error.rs [hunk 1]
```diff
@@ -56,6 +56,10 @@ pub enum WifiConnectionError {
     WaitingForWifiDeactivation,
     BufferOverflow,
     // SsidNotFound,
+    /// Config id above 9.
+    IncompatiableConfigId,
+    /// Trying to enter an illigal state
+    Illigal,
```
Comment (MathiasKoch) [inline ublox-short-range/src/error.rs:62]:
```suggestion
    IncompatibleConfigId,
    /// Trying to enter an illegal state
    Illegal,
```

File: ublox-short-range/src/wifi/supplicant.rs [hunk 2]
```diff
@@ -142,11 +177,19 @@ where
...
 
+    /// Attempts to remove a stored wireless network
+    ///
+    /// Removing the active connection is not possible. Deactivate the network first.
     pub fn remove_connection(&mut self, config_id: u8) -> Result<(), WifiConnectionError> {
-        self.deactivate(config_id)?;
+        // self.deactivate(config_id)?;
+        if let Some(w) = self.wifi_connection {
+            if w.config_id == config_id && w.active {
+                return Err(WifiConnectionError::Illigal);
```
Comment (MathiasKoch) [inline ublox-short-range/src/wifi/supplicant.rs:187]:
```suggestion
                return Err(WifiConnectionError::Illegal);
```

File: ublox-short-range/src/wifi/supplicant.rs [hunk 3]
```diff
@@ -160,117 +203,148 @@ where
...
 
         Ok(())
     }
 
+    /// Activate a given network config
+    /// Only one config can be active at any time.
     pub fn activate(&mut self, config_id: u8) -> Result<(), WifiConnectionError> {
+        if let Some(w) = self.wifi_connection {
+            if w.active {
+                return Err(WifiConnectionError::Illigal);
```
Comment (MathiasKoch) [inline ublox-short-range/src/wifi/supplicant.rs:295]:
```suggestion
                return Err(WifiConnectionError::Illegal);
```

File: ublox-short-range/src/wifi/supplicant.rs [hunk 4]
```diff
@@ -291,4 +365,8 @@ where
             .map(WifiConnection::is_connected)
             .unwrap_or_default()
     }
+
+    pub fn flush(&mut self) -> Result<(), WifiConnectionError> {
+        todo!()
```
Comment (MathiasKoch) [inline ublox-short-range/src/wifi/supplicant.rs:370]: Is this intended?

File: ublox-short-range/src/error.rs [hunk 5]
```diff
@@ -56,6 +56,10 @@ pub enum WifiConnectionError {
     WaitingForWifiDeactivation,
     BufferOverflow,
     // SsidNotFound,
+    /// Config id above 9.
+    IncompatiableConfigId,
```
Comment (MathiasKoch) [inline ublox-short-range/src/error.rs:60]: This error kind seems unused? Is it intended to be used later, or is there something missing in this PR?

**Review by MathiasKoch — CHANGES_REQUESTED** [review]
Minor stuff.

Please update, check that it still compiles and works, and then feel free to squash and merge afterwards, so you can update your follow-up PR's.

**Comment by MathiasKoch** [comment]
@unizippro

File: ublox-short-range/src/error.rs (hunk 5 above)
Comment (unizippro) [reply ublox-short-range/src/error.rs:60]: It is for later use

File: ublox-short-range/src/wifi/supplicant.rs (hunk 4 above)
Comment (unizippro) [reply ublox-short-range/src/wifi/supplicant.rs:370]: The intention is a function removing all stored credentials, as it is not being used the writing of it has been postphoned to https://github.com/BlackbirdHQ/ublox-short-range-rs/pull/51.

**Comment by MathiasKoch** [comment]
LGTM :+1:

**Pull request merged**